code/game/machinery/vendors/custom_vendors.dm

```diff
@@ -30,9 +37,25 @@
 		return ITEM_INTERACT_COMPLETE
 	else if(locked())
 		return ..()
-	if(!user.canUnEquip(used, FALSE))
+
+	try_add_stock(user, used)
+	return ITEM_INTERACT_COMPLETE
+
+/// Tries to add something to the vendor. can_wait returns INSERT_NEEDS_INPUT if it would wait for user input, quiet suppresses success messages, and bag is used when the item is being transferred from a storage item.
+/obj/machinery/economy/vending/custom/proc/try_add_stock(mob/living/user, obj/item/used, can_wait = TRUE, quiet = FALSE, obj/item/storage/bag = null)
+	if(isnull(bag) && !user.canUnEquip(used, FALSE))
 		to_chat(user, "<span class='warning'>\The [used] is stuck to your hand!</span>")
```

Comment:
```suggestion
		to_chat(user, "<span class='warning'>[used] is stuck to your hand!</span>")
```

Reply:
Unlike, `\the`, `\The` is actually useful: It changes capitalization.

Without:
`the hamburger is stuck to your hand!`

With:
`The hamburger is stuck to your hand!`

Reply:
Oh, I didn't actually know this! Sounds good!